modules/business-text/business-text.service.js

```diff
@@ -37,32 +39,122 @@ class BusinessTextService {
       };
     }
 
-    const text = await BusinessText.findByIdAndUpdate(id, businessText, {
+    const newPage = files.length
+      ? await this.pageImageReplaceHandler(businessText, files)
+      : businessText;
+    const oldPage = await this.getBusinessTextById(id);
+
+    if (oldPage) {
+      const newImages = this.imageFinder(newPage);
+      const oldImages = this.imageFinder(oldPage);
+
+      const imagesToDelete = oldImages.filter(
+        img => newImages.findIndex(newImg => newImg === img) === -1
+      );
+
+      if (imagesToDelete.length) {
+        await this.deleteNoNeededImages(imagesToDelete);
+      }
+    } else {
+      throw new Error(BUSINESS_TEXT_NOT_FOUND);
+    }
+
+    const text = await BusinessText.findByIdAndUpdate(id, newPage, {
```

Comment:
If you retrieve the business-text on the beginning of manipulation with it then it is better to use just .save() on existing in the memory business-text object

Reply:
But I need to replace old data to updated one. How it is possible to do on this way?